src/module/actor/lancer-actor.ts

```diff
@@ -243,17 +252,27 @@ export class LancerActor extends Actor {
    * Performs structure on the mech
    * For now, just rolls on table. Eventually we can include configuration to do automation
    */
-  async structure() {
+  async structure(reroll_data?: { structure: number }) {
     // Assert that we're on a mech or NPC
-    if (this.is_mech() || this.is_npc()) {
-      this.structureMech();
-    } else {
+    if (!this.is_mech() && !this.is_npc()) {
       ui.notifications!.warn("Can only structure NPCs and Mechs");
       return;
     }
+    const ent = await this.data.data.derived.mm_promise;
+    if (getAutomationOptions().structure && !reroll_data) {
+      if (ent.CurrentHP < 1 && ent.CurrentStructure >= 0) {
+        ent.CurrentHP += ent.MaxHP;
```

Comment:
A slight bug here shown by the macro firing automatically - if `ent.CurrentStructure == 1` then we shouldn't add max HP, because the guy's dead. Similar for overheat, I think we shouldn't subtract heat if `ent.CurrentStress == 1`.